app-backend/db/src/main/scala/com/azavea/rf/database/ObjectPermissions.scala

```diff
@@ -0,0 +1,98 @@
+  }
+
+  def listUserActionsF(user: User, id: UUID, groupIdsF: String): Fragment =
+    Fragment.const(s"SELECT a.acrs from (SELECT UNNEST(acrs) acrs from ${tableName}") ++
+      Fragments.whereAndOpt(Some(fr"id=${id}")) ++ Fragment.const(") a") ++
+      Fragment.const(s"WHERE a.acrs LIKE '%${user.id}%' OR a.acrs LIKE '%ALL%' OR ${groupIdsF}")
+
+  def acrStringsToList(acrs: List[String]): List[Option[ObjectAccessControlRule]] =
+    acrs.map(ObjectAccessControlRule.fromObjAcrString)
+
+  def getPermissions(id: UUID): ConnectionIO[List[Option[ObjectAccessControlRule]]] = for {
+    isValidObject <- isValidObject(id)
+    getPermissions <- isValidObject match {
+      case false => throw new Exception(s"Invalid ${tableName} object ${id}")
+      case true => getPermissionsF(id).query[List[String]].unique.map(acrStringsToList(_))
+    }
+  } yield { getPermissions }
+
+  def addPermission(id: UUID, acr: ObjectAccessControlRule): ConnectionIO[List[Option[ObjectAccessControlRule]]] = for {
+    permissions <- getPermissions(id)
+    permExists = permissions.contains(Some(acr))
+    addPermission <- permExists match {
+      case true => throw new Exception(s"${acr.toObjAcrString} exists for ${tableName} ${id}")
+      case false => appendPermissionF(id, acr).update.withUniqueGeneratedKeys[List[String]]("acrs").map(acrStringsToList(_))
+    }
+  } yield { addPermission }
+
+  def addPermissionsMany(id: UUID, acrList: List[ObjectAccessControlRule]): ConnectionIO[List[Option[ObjectAccessControlRule]]] = for {
+    permissions <- getPermissions(id)
+    acrListFiltered = acrList.filter(acr => !permissions.contains(Some(acr)))
+    addPermissionsMany <- acrListFiltered.length match {
+      case 0 => throw new Exception(s"All permissions exist for ${tableName} ${id}")
+      case _ => updatePermissionsF(id, acrListFiltered).update.withUniqueGeneratedKeys[List[String]]("acrs").map(acrStringsToList(_))
+    }
+  } yield { addPermissionsMany }
+
+  def replacePermissions(id: UUID, acrList: List[ObjectAccessControlRule]): ConnectionIO[List[Option[ObjectAccessControlRule]]] =
+    updatePermissionsF(id, acrList, true).update.withUniqueGeneratedKeys[List[String]]("acrs").map(acrStringsToList(_))
+
+  def deletePermissions(id: UUID): ConnectionIO[List[Option[ObjectAccessControlRule]]] =
+    replacePermissions(id, List[ObjectAccessControlRule]())
+
+  def listUserActions(user: User, id: UUID): ConnectionIO[List[String]] = for {
+    ugrs <- UserGroupRoleDao.listByUser(user)
+    groupIdString = ugrs.map((urg: UserGroupRole) => s"a.acrs LIKE '%${urg.groupId.toString}%'").mkString(" OR ")
+    listUserActions <- listUserActionsF(user, id, groupIdString).query[String].to[List]
+    actions = acrStringsToList(listUserActions).flatten.map(_.actionType.toString).distinct
+  } yield { actions }
+
+  // function not used anywhere
+  // def listBySubject(subjectType: SubjectType, subjectId: String)
+
+  // todo in a future card
```

Comment:
I would leave out the TODO, and just open an issue that describes that we need a batch command to delete all team ACRs in an asynchronous way